static/src/javascripts/projects/commercial/modules/messenger/background.js

```diff
@@ -38,50 +43,99 @@ const setBackground = (specs: AdSpec, adSlot: Node): Promise<any> => {
         !('backgroundImage' in specs) ||
         !('backgroundRepeat' in specs) ||
         !('backgroundPosition' in specs) ||
-        !('scrollType' in specs)
+        !('scrollType' in specs) ||
+        !(adSlot instanceof Element)
     ) {
         return Promise.resolve();
     }
 
     const specStyles: SpecStyles = getStylesFromSpec(specs);
 
-    // Create an element to hold the background image
-    const background = document.createElement('div');
-    background.className = `creative__background creative__background--${specs.scrollType}`;
-    Object.assign(background.style, specStyles);
-
-    // Wrap the background image in a DIV for positioning. Also, we give
-    // this DIV a background colour if it is provided. This is because
-    // if we set the background colour in the creative itself, the background
-    // image won't be visible (think z-indexed layers)
-    const backgroundParent = document.createElement('div');
-    backgroundParent.className = 'creative__background-parent';
-    if (specStyles.backgroundColor) {
-        backgroundParent.style.backgroundColor = specStyles.backgroundColor;
-    }
-    backgroundParent.appendChild(background);
+    // check to see whether the parent div exists already, if so, jut alter the style
+    const backgroundParentClass = 'creative__background-parent';
+    const backgroundClass = 'creative__background';
+
+    const maybeBackgroundParent = adSlot.querySelector(
+        `.${backgroundParentClass}`
+    );
+    const maybeBackground = maybeBackgroundParent
+        ? maybeBackgroundParent.querySelector(`.${backgroundClass}`)
```

Comment:
If you'd use `maybeBackgroundParent.getElementsByClassName(backgroundClass)` you don't need the string concatenation. It's also faster and @GHaberis would have suggested it anyways ;)

Reply:
Ah good to know! I'll make that change; I got a bit in the weeds when trying to convince flow that certain methods would exist on my objects (mainly because passing around `Node` is so entrenched in this area of the commercial stack rather than `HTMLElement`) and so settled on `querySelector`. 👍